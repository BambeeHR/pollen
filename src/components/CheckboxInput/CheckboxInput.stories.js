import BaseIcon from '../BaseIcon/BaseIcon.vue';
import Form from '../../constants/Form';
import CheckboxInput from './CheckboxInput.vue';
import TypeOverline from '../TypeOverline/TypeOverline.vue';
import '../../assets/tailwind.css';

const { Sizes, Variants } = Form;
const sizes = [Sizes.DENSE, Sizes.NORMAL];

export default {
  title: 'Toggles/CheckboxInput',
  component: CheckboxInput,
  story: {
    parameters: {
      jest: ['CheckboxInput.test.js'],
    },
  },
};

export const Gallery = () => ({
  components: { CheckboxInput, TypeOverline },
  data() {
    return {
      sizes,
      value: true,
      variants: [Variants.STANDARD, Variants.RAISED],
    };
  },
  template: `
  <div class="flex -mx-8">
    <div class="flex-grow mx-8" v-for="variant in variants" :key="variant">
      <TypeOverline tag="h2" variant="large" class="mb-2">{{ variant }}</TypeOverline>
      <div v-for="size in sizes" :key="variant + '-' + size">
        <TypeOverline tag="h3" class="mb-2">{{ size }}</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" label="My checkbox" />
        <TypeOverline tag="h3" class="mb-2">{{ size }}, invalid</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" label="My checkbox" invalid />
        <TypeOverline tag="h3" class="mb-2">{{ size }}, disabled</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" label="My checkbox" disabled />
      </div>
    </div>
  </div>
  `,
});

export const LabelOnTheLeft = () => ({
  components: { CheckboxInput, TypeOverline },
  data() {
    return {
      sizes,
      value: true,
      variants: [Variants.STANDARD, Variants.RAISED],
    };
  },
  template: `
  <div class="flex -mx-8">
    <div class="flex-grow mx-8" v-for="variant in variants" :key="variant">
      <TypeOverline tag="h2" variant="large" class="mb-2">{{ variant }}</TypeOverline>
      <div v-for="size in sizes" :key="variant + '-' + size">
        <TypeOverline tag="h3" class="mb-2">{{ size }}</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" label="My checkbox" label-position="left" />
        <TypeOverline tag="h3" class="mb-2">{{ size }}, invalid</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" label="My checkbox" label-position="left" invalid />
        <TypeOverline tag="h3" class="mb-2">{{ size }}, disabled</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" label="My checkbox" label-position="left" disabled />
      </div>
    </div>
  </div>
  `,
});

export const HTMLLabel = () => ({
  components: { CheckboxInput, TypeOverline },
  data() {
    return {
      sizes: Object.values(Sizes),
      value: true,
      variants: [Variants.STANDARD, Variants.RAISED],
    };
  },
  template: `
  <div class="flex -mx-8">
    <div class="flex-grow mx-8" v-for="variant in variants" :key="variant">
      <TypeOverline tag="h2" variant="large" class="mb-2">{{ variant }}</TypeOverline>
      <div v-for="size in sizes" :key="variant + '-' + size">
        <TypeOverline tag="h3" class="mb-2">{{ size }}</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" label="My <i>checkbox</i>" />
        <TypeOverline tag="h3" class="mb-2">{{ size }}, invalid</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" label="My <i>checkbox</i>" invalid />
        <TypeOverline tag="h3" class="mb-2">{{ size }}, disabled</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" label="My <i>checkbox</i>" disabled />
      </div>
    </div>
  </div>
  `,
});

export const WithLabelAsSlot = () => ({
  components: { BaseIcon, CheckboxInput, TypeOverline },
  data() {
    return {
      sizes: Object.values(Sizes),
      value: true,
      variants: [Variants.STANDARD, Variants.RAISED],
    };
  },
  template: `
  <div class="flex -mx-8">
    <div class="flex-grow mx-8" v-for="variant in variants" :key="variant">
      <TypeOverline tag="h2" variant="large" class="mb-2">{{ variant }}</TypeOverline>
      <div v-for="size in sizes" :key="variant + '-' + size">
        <TypeOverline tag="h3" class="mb-2">{{ size }}</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size">
          <div>
            <BaseIcon icon="lock" />
            Lock it down?
          </div>
        </CheckboxInput>
        <TypeOverline tag="h3" class="mb-2">{{ size }}, invalid</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" invalid>
          <div>
            <BaseIcon icon="lock" />
            Lock it down?
          </div>
        </CheckboxInput>
        <TypeOverline tag="h3" class="mb-2">{{ size }}, disabled</TypeOverline>
        <CheckboxInput v-model="value" :variant="variant" :size="size" disabled>
          <div>
            <BaseIcon icon="lock" />
            Lock it down?
          </div>
        </CheckboxInput>
      </div>
    </div>
  </div>
  `,
});

/* eslint-disable no-param-reassign */
[Gallery, LabelOnTheLeft, HTMLLabel, WithLabelAsSlot].forEach((item) => {
  item.story = {
    parameters: {
      jest: ['CheckboxInput.test.js'],
    },
  };
});
