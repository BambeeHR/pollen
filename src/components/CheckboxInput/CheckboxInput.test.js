import { mount } from '@vue/test-utils';
import CheckboxInput from './CheckboxInput.vue';

describe('CheckboxInput', () => {
  test('renders correctly', () => {
    const wrapper = mount(CheckboxInput, {
      propsData: {
        value: true,
        id: 'test',
        label: 'test',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('emits an input event', async () => {
    const Component = {
      components: { CheckboxInput },
      template: `
        <div>
          <CheckboxInput v-model="value" label="test" />
        </div>
      `,
      data() {
        return {
          value: false,
        };
      },
    };
    const wrapper = mount(Component);
    const input = wrapper.find('input');
    await input.trigger('click');
    expect(wrapper.vm.value).toEqual(true);
    await input.trigger('click');
    expect(wrapper.vm.value).toEqual(false);
  });

  test('emits an input event with defined trueValue and falseValue prop strings', async () => {
    const Component = {
      components: { CheckboxInput },
      template: `
        <div>
          <CheckboxInput v-model="value" true-value="yes" false-value="no" label="test" />
        </div>
      `,
      data() {
        return {
          value: 'yes',
        };
      },
    };
    const wrapper = mount(Component);
    const input = wrapper.find('input');
    await input.trigger('click');
    expect(wrapper.vm.value).toEqual('no');
    await input.trigger('click');
    expect(wrapper.vm.value).toEqual('yes');
  });
});
