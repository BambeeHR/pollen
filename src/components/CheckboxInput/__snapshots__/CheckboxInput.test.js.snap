// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxInput renders correctly 1`] = `
<label
  class="toggle-input checkbox-input toggle-input--normal toggle-input--standard toggle-input--label-right"
  for="test"
>
  <input
    class="toggle-input__input"
    id="test"
    name="test"
    type="checkbox"
  />
   
  <div
    class="checkbox-indicator checkbox-indicator--normal checkbox-indicator--standard checkbox-indicator--checked"
  >
    <svg
      aria-hidden="true"
      class="base-icon svg-inline--fa fa-check fa-w-16"
      data-icon="check"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class=""
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </div>
   
  <div
    class="toggle-input__label"
  >
    <span>
      test
    </span>
  </div>
</label>
`;
